src/doom/r_segs.c

```diff
@@ -87,7 +87,47 @@ lighttable_t**	walllights;
 
 short*		maskedtexturecol;
 
+//
+// [kg] medusa effect indicator
+extern int *texturecompositesize;
```

Comment:
Don't put `extern` in a .c file. Put it in an .h file where it will also reach the actual definition. This way the compiler will check it's correct.